packages/player/src/components/hls-source.tsx

```diff
@@ -1,31 +1,80 @@
-import React, {Component} from 'react'
+import * as React from 'react'
 import Hls from 'hls.js'
+import {VideoContext} from '../context/video-context'
+import {useSelector} from '@xstate/react'
+import {selectVideo} from './player'
 
-export default class HLSSource extends Component<any> {
-  hls
+type HLSSourceProps = {
+  hlsConfig?: any
+  src: string
+  type?: string
+}
+
+/**
+ * Wrapper for the hls.js library that attaches to an active video element
+ * and allows us to load m3u8 (hls) streaming video
+ *
+ * @see {@link https://github.com/devcshort/react-hls/blob/master/src/index.tsx}
```

Comment:
almost a complete cut and paste from this tidy little implementation loading HLS in a useEffect hook